### ISSUE TITLE
Pyright giving type errors after updating to nextcord 2.4.0

### ISSUE THREAD

**Issue by n0tcnj1eck4h** [nextcord/nextcord#997]
### Summary

After upgrading from nextcord 2.3.3 to 2.4.0 pyright shows a long type error on every command decorator inside a cog

### Reproduction Steps

- Create a class deriving from commands.Cog
- Create a command using the commands.commands decorator
- Use Pyright
```
Pyright: Argument of type "(self: Self@ExampleCog, ctx: Context[Unknown]) -> Coroutine[Any, Any, None]" cannot be assigned to parameter of type "((CogT, ContextT@command, **P) -> Coro[T]) | ((ContextT@command, **P) -> Coro[T])"
  Type "(self: Self@ExampleCog, ctx: Context[Unknown]) -> Coroutine[Any, Any, None]" cannot be assigned to type "((CogT, ContextT@command, **P) -> Coro[T]) | ((ContextT@command, **P) -> Coro[T])"
    Type "(self: Self@ExampleCog, ctx: Context[Unknown]) -> Coroutine[Any, Any, None]" cannot be assigned to type "(CogT, ContextT@command, **P) -> Coro[T]"
      Parameter 1: type "CogT" cannot be assigned to type "Self@ExampleCog"
        "Cog*" is incompatible with "ExampleCog"
    Type "(self: Self@ExampleCog, ctx: Context[Unknown]) -> Coroutine[Any, Any, None]" cannot be assigned to type "(ContextT@command, **P) -> Coro[T]"
      Parameter 1: type "ContextT@command" cannot be assigned to type "Self@ExampleCog"
        "Context[Unknown]*" is incompatible with "ExampleCog"
```

### Minimal Reproducible Code

```python
from nextcord.ext import commands

class ExampleCog(commands.Cog):
    @commands.command()
    async def example_command(self, ctx: commands.Context):
        pass
```

### Expected Results

No type error

### Actual Results

Overwhelming type error

### Intents

unrelated

### System Information

Linux 6.1.12-arch1-1
pyright 1.1.294
nextcord==2.4.0

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

Video showcasing the behavior:

https://user-images.githubusercontent.com/48856951/220776037-486b9d4f-1178-44ff-952e-a68d3632780d.mp4

[Affected repository](https://github.com/skni-kod/PyChan-Bot/tree/develop)

**Comment by teaishealthy** [comment]
Can repro

**Comment by teaishealthy** [comment]
Type regression from #894

**Comment by EmreTech** [comment]
This issue has been fixed in v2.5 (01d4ef0a05b3c2c83426dd8228f26cb65a77bc16).
